src/core/Bootstrapper.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable no-plusplus */
+/* eslint no-await-in-loop: 0 */
+
+export default class Bootstrapper {
+
+  /**
+     *
+     * @param {RCISDKConfig} config
+     */
+  constructor (config, transport, producer) {
+    this.config = config;
+    this.transport = this.provisionTransport(transport);
+    this.producer = this.provisionProducer(producer);
+  }
+
+  provisionTransport (Transport) {
+    let input;
+    if ((Array.isArray(this.config.tenancies) && this.config.tenancies.length > 0) && ((typeof this.config.targetUrl === 'string' || this.config.targetUrl instanceof String) && this.config.targetUrl.length > 0)) {
+      // If tenancies is an populated array and targetUrl is a populated string, then choose to provision the transport in the new way
+      input = this.config.tenancies;
+    } else if (Array.isArray(this.config.tenancies) && this.config.tenancies.length > 0) {
```

Comment:
@robertdumitrescu KISS - This logical branch appears the same as the one above - I don't understand what is different..?